### PR TITLE
Add linux -> windows cross-compilation

### LINKED ISSUE #183: polars >= 0.19.7 not buildable on windows azure runners.
As observed during #182 newer versions of Polars require significantly more memory to build compared to older versions. This has led to OOM errors on Azure Pipeline runners for both Linux and Windows.

On Linux, we can mitigate this issue by enabling swapping, but the Windows environment is more challenging to address. Adjusting the system pagefile on Windows doesn't take effect until after a system reboot, which is not practical in this context.

Previous hotfixes (https://github.com/conda-forge/polars-feedstock/commit/e88541f2933153187aef3cccaa52bce0c8a5fccf, https://github.com/conda-forge/polars-feedstock/commit/fb518ca24bf6b174f468afe991ad342cd94417e0) attempted to reduce memory requirements, particularly during the linking phase, by lowering the LTO level. 

One option is to disable LTO entirely on Windows to atleast get Windows building again, although this will almost certainly result in a performance hit.

Another option would be to try to compile Polars for Windows using a Linux runner as a build platform.

Suggestions and insights are welcome.

### PR DESCRIPTION
**Pull request by pavelzw**
Closes #183 

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,27 +10,28 @@ source:
     sha256: 7ba32d104832fed1ac75b8f264f11953751edf062b2ab9d4e69b19ab5a224a0c
 
 build:
-  number: 0
-  skip: true  # [win]
-  skip: true  # [py<37]
+  number: 1
   skip: true  # [win and python_impl=="pypy"]
```
Comment (pavelzw) [inline recipe/meta.yaml:14]:
```suggestion
  skip: true  # [win and python_impl=="pypy"]
  skip: true  # [target_platform != "win-64"]
```

**Comment by pavelzw** [comment]
@conda-forge-admin, please rerender

**Comment by 0xbe7a** [comment]
depends on https://github.com/conda-forge/maturin-feedstock/pull/109 and https://github.com/conda-forge/rust-activation-feedstock/pull/38

**Comment by 0xbe7a** [comment]
@conda-forge-admin, please restart ci

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,27 +10,26 @@ source:
...
     - python                              # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}  # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}  # [build_platform != target_platform and not (build_platform == "linux-64" and target_platform == "win-64")]
     - crossenv                            # [build_platform != target_platform]
     - maturin >=1.2.1,<2                  # [build_platform != target_platform]
-    - {{ compiler('c') }}
+    - {{ compiler('c') }}                 # [not (build_platform == "linux-64" and target_platform == "win-64")]
     - {{ compiler('rust') }}
-    - posix                               # [win]
+    - posix                               # [win and not (build_platform == "linux-64" and target_platform == "win-64")]
```
Comment (xhochy) [inline recipe/meta.yaml:24]:
```suggestion
    - posix                               # [build_platform == "win-64"]
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -10,27 +10,26 @@ source:
...
-  skip: true  # [win]
-  skip: true  # [py<37]
+  number: 1
   skip: true  # [win and python_impl=="pypy"]
 
 requirements:
   build:
     - python                              # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}  # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}  # [build_platform != target_platform and not (build_platform == "linux-64" and target_platform == "win-64")]
```
Comment (xhochy) [inline recipe/meta.yaml:19]: Can you add a comment why this and `compiler('c')` is excluded in the win-cross situation?

**Review by xhochy — CHANGES_REQUESTED** [review]
Before we merge, we should definitely rerender this to ensure that nothing breaks other systems.

**Comment by pavelzw** [comment]
You can install the builds from [`45450e1` (#187)](https://github.com/conda-forge/polars-feedstock/pull/187/commits/45450e1d3d74787812092f215fadc1e80d73801d) using

```
micromamba install -c https://repo.prefix.dev/test-polars polars=0.19.11
```

**Comment by pavelzw** [comment]
I did some light testing with the cross-compiled build and everything seems to work 👍🏻

**Comment by xhochy** [comment]
We will need https://github.com/conda-forge/clang-win-activation-feedstock/pull/31 to get rid of 

```
2023-10-26T13:47:26.8314466Z WARNING (polars,lib/site-packages/polars/polars.pyd): Needed DSO api-ms-win-crt-stdio-l1-1-0.dll found in ['ucrt']
2023-10-26T13:47:26.8319964Z WARNING (polars,lib/site-packages/polars/polars.pyd): .. but ['ucrt'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
2023-10-26T13:47:26.8322721Z WARNING (polars,lib/site-packages/polars/polars.pyd): Needed DSO api-ms-win-crt-heap-l1-1-0.dll found in ['ucrt']
2023-10-26T13:47:26.8324106Z WARNING (polars,lib/site-packages/polars/polars.pyd): .. but ['ucrt'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
2023-10-26T13:47:26.8330566Z WARNING (polars,lib/site-packages/polars/polars.pyd): Needed DSO api-ms-win-crt-runtime-l1-1-0.dll found in ['ucrt']
2023-10-26T13:47:27.8297454Z WARNING (polars,lib/site-packages/polars/polars.pyd): .. but ['ucrt'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
2023-10-26T13:47:27.8299486Z WARNING (polars,lib/site-packages/polars/polars.pyd): Needed DSO api-ms-win-crt-convert-l1-1-0.dll found in ['ucrt']
2023-10-26T13:47:27.8306434Z WARNING (polars,lib/site-packages/polars/polars.pyd): .. but ['ucrt'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
2023-10-26T13:47:27.8307935Z WARNING (polars,lib/site-packages/polars/polars.pyd): Needed DSO api-ms-win-crt-string-l1-1-0.dll found in ['ucrt']
2023-10-26T13:47:27.8309333Z WARNING (polars,lib/site-packages/polars/polars.pyd): .. but ['ucrt'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
2023-10-26T13:47:27.8313485Z WARNING (polars,lib/site-packages/polars/polars.pyd): Needed DSO api-ms-win-crt-math-l1-1-0.dll found in ['ucrt']
2023-10-26T13:47:27.8314925Z WARNING (polars,lib/site-packages/polars/polars.pyd): .. but ['ucrt'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
```

**Review by xhochy — APPROVED** [review]

**Comment by 0xbe7a** [comment]
@conda-forge-admin, please restart ci

**Pull request merged**